File: designtime/webstudio/ui/src/app/editors/rule-template-instance-view/BindingInfo.ts
/**
 * @Author: Rahil Khera
 * @Date:   2017-04-20T10:08:06+05:30
 * @Last modified by:   Rahil Khera
 * @Last modified time: 2017-04-20T10:08:07+05:30
 */

import { DomainInfo } from './DomainInfo';

export class BindingInfo {

  private bindingId: string = undefined;
  private type: string = undefined;
  private value: string = undefined;
  private domainInfo: DomainInfo = null;
  private style: string = undefined;

  constructor(id?: string, type?: string, value?: string, domainInfo?: DomainInfo) {
    this.bindingId = id;
    this.type = type;
    this.value = value;
    this.domainInfo = domainInfo;
    this.style = '';
  }

  setBindingId(id: string) {
    this.bindingId = id;
  }

  getBindingId(): string {
    return this.bindingId;
  }

  setType(type: string) {
    this.type = type;
  }

  getType(): string {
    return this.type;
  }

  setValue(value: string) {
    this.value = value;
  }

  getValue(): string {
    return this.value;
  }

  setDomainInfo(domainInfo: DomainInfo) {
    this.domainInfo = domainInfo;
  }

  getDomainInfo(): DomainInfo {
    return this.domainInfo;
  }

  setStyle(style: string) {
    this.style = style;
  }

  getStyle(): string {
    return this.style;
  }
}

File: designtime/webstudio/WebStudio/src/com/tibco/cep/webstudio/client/model/ruletemplate/BindingInfo.java
package com.tibco.cep.webstudio.client.model.ruletemplate;

import java.io.Serializable;

/**
 * Model for storing Rule Template Instance Bindings
 * 
 * @author Vikram Patil
 */
public class BindingInfo implements Serializable {

	private static final long serialVersionUID = 1L;
	
	private String id;
	private String type;
	private String value;
	private DomainInfo domainInfo;
	private String style;

	public BindingInfo() {
		super();
	}

	public BindingInfo(String id, String type, String value, DomainInfo domainInfo) {
		this.setId(id);
		this.setType(type);
		this.setValue(value);
		this.setDomainInfo(domainInfo);
		this.setStyle("");
	}

	public String getId() {
		return this.id;
	}

	public void setId(String type) {
		this.id = type;
	}

	public DomainInfo getDomainInfo() {
		return this.domainInfo;
	}

	public void setDomainInfo(DomainInfo domainInfo) {
		this.domainInfo = domainInfo;
	}

	public void setType(String type) {
		this.type = type;
	}

	public String getType() {
		return this.type;
	}

	public void setValue(String value) {
		this.value = value;
	}

	public String getValue() {
		return this.value;
	}

	public String getStyle() {
		return this.style;
	}

	public void setStyle(String style) {
		this.style = style;
	}

}
